### PR TITLE
Refactor auth

### PR DESCRIPTION
**Pull request by dtsehl**
### Why is this pull request necessary?
It abstracts away authorization to its own methods and classes.

### How does it fix the issue?
SRP and added tests.

### What side effects might this have?
None - the app is better!

### CODE REVIEW

**Review by jtrtj — APPROVED** [review]
:shipit:

File: .github/workflows/ruby.yml [hunk 1]
```diff
@@ -20,14 +20,16 @@ jobs:
...
-    - name: Set up Ruby
-    # To automatically get bug fixes and new Ruby versions for ruby/setup-ruby,
-    # change this to (see https://github.com/ruby/setup-ruby#versioning):
-    # uses: ruby/setup-ruby@v1
-      uses: ruby/setup-ruby@v1
-      with:
-        ruby-version: 2.7.1
-    - name: Install dependencies
-      run: bundle install
+    - name: Setup ENV
```
Comment (dtsehl) [inline .github/workflows/ruby.yml:23]: This was an interesting one to get set up - a bit of knowledge of Linux commands go a long way!

File: app/api.rb [hunk 2]
```diff
@@ -1,23 +1,26 @@
...
-
 require_relative 'models/user'
+require_relative 'decorators/user'
 
 module Conduit
   class API < Grape::API
     prefix :api
     format :json
 
+    helpers do
```
Comment (dtsehl) [inline app/api.rb:10]: This Controller file now uses helper methods and models/decorators/services to do the actions that it did before itself.

File: db/database.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+require 'sequel'
+require 'pg'
+require 'singleton'
+
+class Database
```
Comment (dtsehl) [inline db/database.rb:5]: In the future we can refactor this class a bit and use a Rake task to automate some of its functionality.

**Pull request merged**